api/src/main/java/org/apache/iceberg/metrics/ScanReport.java

```diff
@@ -117,12 +149,285 @@ public Builder fromScanMetrics(ScanMetrics newScanMetrics) {
       return this;
     }
 
+    public Builder fromScanMetricsResult(ScanMetricsResult newScanMetricsResult) {
+      this.scanMetricsResult = newScanMetricsResult;
+      return this;
+    }
+
     public ScanReport build() {
       Preconditions.checkArgument(null != tableName, "Invalid table name: null");
       Preconditions.checkArgument(null != filter, "Invalid expression filter: null");
       Preconditions.checkArgument(null != projection, "Invalid schema projection: null");
+      Preconditions.checkArgument(
+          null != scanMetrics || null != scanMetricsResult, "Invalid scan metrics: null");
+      return new ScanReport(
+          tableName,
+          snapshotId,
+          filter,
+          projection,
+          null != scanMetrics ? ScanMetricsResult.fromScanMetrics(scanMetrics) : scanMetricsResult);
```

Comment:
Wouldn't it be simpler to convert `scanMetrics` to a result in `fromScanMetrics` and have just `scanMetricsResult` to deal with in this method?

Reply:
good point, that is indeed simpler.